src/lib/ndpi_main.c

```diff
@@ -1700,11 +1700,9 @@ u_int8_t ndpi_is_tor_flow(struct ndpi_detection_module_struct *ndpi_struct,
   struct ndpi_packet_struct *packet = &flow->packet;
 
   if(packet->tcp != NULL) {
-    if(flow->packet.iph) {
-      struct in_addr saddr = { packet->iph->saddr };
-      struct in_addr daddr = { packet->iph->daddr };
-      if(tor_ptree_match(ndpi_struct, &saddr)
-         || tor_ptree_match(ndpi_struct, &daddr)) {
+    if(packet->iph) {
+      if(tor_ptree_match(ndpi_struct, (struct in_addr *)&packet->iph->saddr)
```

Comment:
Since we are not writing to `pin` anymore we can pass the iph field in directly, no need to extract it first anymore.